pkg/types/chain_writer.go

```diff
@@ -11,7 +11,7 @@ type ChainWriter interface {
 	// SubmitSignedTransaction packs and broadcasts a transaction to the underlying chain.
 	//
 	// The `transactionID` will be used by the underlying TXM as an idempotency key, and unique reference to track transaction attempts.
-	SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
+	SubmitSignedTransaction(ctx context.Context, args []any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) error
```

Comment:
Let's also rename this to `CreateTransaction`

Reply:
Mh. I'm not sure about the rename. We're wrapping `CreateTransaction`, but this method only prepares and submits the transaction to the relevant TXM.